lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+/**
+ * TODO: Support batch queries
+ */
+
+const semver = require('semver')
+const sqlSummary = require('sql-summary')
+
+const shimmer = require('../shimmer')
+
+module.exports = function (cassandra, agent, version, enabled) {
+  if (!enabled) return cassandra
+  if (!semver.satisfies(version, '^3.0.0')) {
+    agent.logger.debug('cassandra version %s not supported - aborting...', version)
+    return cassandra
+  }
+
+  if (cassandra.Client) {
+    agent.logger.debug('shimming cassandra.Client.prototype._innerExecute')
+    shimmer.wrap(cassandra.Client.prototype, '_innerExecute', wrapInnerExecute)
+    shimmer.wrap(cassandra.Client.prototype, '_batchCb', wrapBatch)
+  }
+
+  return cassandra
+
+  function wrapBatch (original) {
+    return function wrappedBatch (queries, options, callback) {
+      const spans = []
+
+      for (let query of queries) {
```

Comment:
Seems kind of wasteful to create an array of queries that are all ended an started at the same time. Does the driver send the queries to the database all in the same request, or is it just an abstraction layer where each query is actually performed independently?

Reply:
I'm not really sure how to express the multiple queries in a single span. Ideas?

Reply:
Do you know if the batch of queries are sent to the database in one request? If so why not simply have a single span? Splitting up the spans seem to indicate multiple requests to the database.

I guess your question is how we then record the query string. This is a bit of a hack, but how about joining all query strings with `;\n` and reporting the resulting string on the batch span? I don't like that solution, but I think I like it more than creating one span per query in the batch. But I might be viewing this from the wrong angle. So I can probably be convinced otherwise.

Reply:
Well, they kind of come back off a stream, internally. So it _theoretically_ "starts" and "finishes" the query at a different time, but that's not really usefully visible to the user in any way. Just joining on `;` should be fine. I just wasn't sure if there was any UI implications of doing that, or if we wanted to keep them logically separate for any reason.